Microsoft.Toolkit/Extensions/ISettingsStorageHelperExtensions.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Toolkit.Helpers;
+
+namespace Microsoft.Toolkit.Extensions
+{
+    /// <summary>
+    /// Helpers methods for working with <see cref="ISettingsStorageHelper{TKey}"/> implementations.
+    /// </summary>
+    public static class ISettingsStorageHelperExtensions
+    {
+        /// <summary>
+        /// Attempts to read the provided key and return the value.
+        /// If the key is not found, the fallback value will be used instead.
+        /// </summary>
+        /// <typeparam name="TKey">The type of key used to lookup the object.</typeparam>
+        /// <typeparam name="TValue">The type of object value expected.</typeparam>
+        /// <param name="storageHelper">The storage helper instance fo read from.</param>
+        /// <param name="key">The key of the target object.</param>
+        /// <param name="value">The value of the target object, or the fallback value.</param>
+        /// <param name="fallback">An alternative value returned if the read fails.</param>
+        /// <returns>A boolean indicator of success.</returns>
+        public static bool TryGetValueOrDefault<TKey, TValue>(this ISettingsStorageHelper<TKey> storageHelper, TKey key, out TValue? value, TValue? fallback = default)
```

Comment:
I don't think it makes sense to have a fallback parameter in a method that's already a try-get method.
In case of failure we should simply assign `default` and callers can deal with that 🙂

Reply:
Oh, I left that in there for @michael-hawker's use case in XAML Studio. It's optional, and in an extension method, so I don't think it's too egregious to leave it. Let me know if you feel strongly otherwise.

Reply:
Oh, I thought @michael-hawker was just referring to the fallback parameter in the non-try method? As in, if you just always want a return value and want to specify an explicit fallback, and don't really care whether the value already existed or not, then wouldn't it make more sense to just make this method be:

```csharp
public static TValue GetValueOrDefault<TKey, TValue>(this ISettingsStorageHelper<TKey> storageHelper, TKey key, TValue? fallback = default)
```

My only concern here is that the `Try___` pattern combined with also a fallback parameter is not really idiomatic and it's a bit unusual to have, you'd usually only have one or the other but not both. That said I wouldn't consider this to be a blocking issue at this point, so I'm fine with either leaving this or just changing this to have the signature above 🙂

Reply:
Haha, that's a good point now that you point it out. I'll ditch the `Try` syntax